pkg/lvm/lvm_util.go

```diff
@@ -571,24 +682,87 @@ func getLvDeviceName(path string) (string, error) {
 //It returns LogicalVolume.
 //
 //Example: LogicalVolume{
-//		Name:     "pvc-213ca1e6-e271-4ec8-875c-c7def3a4908d",
-//		Size:     3221225472,
-//		Path:     "/dev/linuxlvmvg/pvc-213ca1e6-e271-4ec8-875c-c7def3a4908d",
+//		Name:               "pvc-082c7975-9af2-4a50-9d24-762612b35f94",
+//		FullName:           "vg_thin/pvc-082c7975-9af2-4a50-9d24-762612b35f94"
+//		UUID:               "FBqcEe-Ln72-SmWO-fR4j-t4Ga-1Y90-0vieKW"
+//		Size:                4294967296,
+//		Path:                "/dev/vg_thin/pvc-082c7975-9af2-4a50-9d24-762612b35f94",
+//		DMPath:              "/dev/mapper/vg_thin-pvc--082c7975--9af2--4a50--9d24--762612b35f94"
+//		Device:              "dm-5"
+//		VGName:              "vg_thin"
+//		SegType:             "thin"
+//		Permission:          1
+//		BehaviourWhenFull:   -1
+//		HealthStatus:        0
+//		RaidSyncAction:      -1
+//		ActiveStatus:        "active"
+//		Host:                "node1-virtual-machine"
+//		PoolName:            "vg_thin_thinpool"
+//		UsedSizePercent:     0
+//		MetadataSize:        0
+//		MetadataUsedPercent: 0
+//		SnapshotUsedPercent: 0
 //	}
 func parseLogicalVolume(m map[string]string) (LogicalVolume, error) {
 	var lv LogicalVolume
 	var err error
+	var sizeBytes int64
+	var count float64
 
 	lv.Name = m["lv_name"]
+	lv.FullName = m["lv_full_name"]
+	lv.UUID = m["lv_uuid"]
 	lv.Path = m["lv_path"]
-	sizeBytes, err := strconv.ParseInt(strings.TrimSuffix(strings.ToLower(m["lv_size"]), "b"), 10, 64)
+	lv.DMPath = m["lv_dm_path"]
+	lv.VGName = m["vg_name"]
+	lv.ActiveStatus = m["lv_active"]
 
-	if err != nil {
-		err = fmt.Errorf("invalid format of lv_size=%v for lv %v: %v", m["lv_size"], lv.Name, err)
-		return LogicalVolume{}, err
+	int64Map := map[string]*int64{
+		"lv_size":          &lv.Size,
+		"lv_metadata_size": &lv.MetadataSize,
+	}
+	for key, value := range int64Map {
+		// Check if the current LV is not a thin pool. If not then
+		// metadata size will not be present as metadata is only
+		// stored for thin pools.
+		if m["segtype"] != LVThinPool && key == "lv_metadata_size" {
```

Comment:
What is the behavior if we dont have this check?

Reply:
It will throw an error. Because metadata field has value only in-case of thin pools. Else it has an empty string value. That can trying to convert to int64 will error out.